src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1541,6 +1541,27 @@ private static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSta
                 else
                 {
                     var property = sourceMethod.AssociatedSymbol as SourcePropertySymbol;
+                    if (!sourceMethod.IsExtern &&
+                        !sourceMethod.IsAbstract &&
+                        !sourceMethod.IsPartial &&
+                        !sourceMethod.IsExpressionBodied &&
```

Comment:
On reading this again, I'm not sure I need to check `IsExpressionBodied` here. I'll try to remove that check.